packages/core/src/root.ts

```diff
@@ -239,6 +252,10 @@ export function createRemoteRoot<
     const normalizedChild =
       typeof child === 'string' ? remoteRoot.createText(child) : child;
 
+    if (!componentsSet.has(normalizedChild)) {
+      throw new Error('Append invalid component');
```

Comment:
I think I'd use a more instructive message here, something like `Can’t append a node that was not created by this RemoteRoot`

Reply:
Fixed!